xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-api/src/main/java/org/xwiki/rest/resources/classes/ClassPropertyValuesProvider.java

```diff
@@ -46,4 +46,18 @@
      */
     PropertyValues getValues(ClassPropertyReference propertyReference, int limit, Object... filterParameters)
         throws XWikiRestException;
+
+    /**
+     * Provides the document specified by the filter as a property value.
+     *
+     * @param propertyReference the property to provide the value for
+     * @param rawValue raw value used to resolve the property value
+     * @return the property value based on the raw value
+     * @throws XWikiRestException if retrieving the property value fails
+     * @since 10.9RC1
+     */
+    default PropertyValues getValue(ClassPropertyReference propertyReference, Object rawValue)
```

Comment:
Why not return ``PropertyValue`` (singular) because the method is supposed to return a single value? The only advantage I see for returning ``PropertyValues`` (plural) is to avoid NPE (you return an empty collection instead of null). But on the other hand it complicates the code a bit because you need to work with a collection. WDYT?

Reply:
I think I did that because the related REST API needed a collection. It is indeed more logical to have the method return a single value or null and let the caller wrap it into a collection.